### PR TITLE
Adds extensive testing and fixes from backlog

### PR DESCRIPTION
**Pull request by mcamarad**
# Summary:

Addresses fixes and improvements from gel-gwas backlog.

# Tasks:

- [x] [plinkTestCh is not used as input anywhere](https://github.com/lifebit-ai/gel-gwas/issues/37)
- [x] [Add tests to cover all run scenaria.](https://github.com/lifebit-ai/gel-gwas/issues/43)
- [x] [Fix null behaviour on report when LDSC is not executed](https://github.com/lifebit-ai/gel-gwas/issues/70)
- [x] [Refactor to have only one process](https://github.com/lifebit-ai/gel-gwas/issues/68)
- [x] [Opportunity for refactor: Flatten nested if (if (case_group != 'None'){](https://github.com/lifebit-ai/gel-gwas/issues/45)
- [x] [Opportunity for refactor: flatten nested ifs](https://github.com/lifebit-ai/gel-gwas/issues/44)
- [x] [Add exception in continuous data transformations to not be computed on PCs](https://github.com/lifebit-ai/gel-gwas/issues/69)
- [x] [Adds testing scenario for working with real pheno data and synthetic genotypes](https://github.com/lifebit-ai/gel-gwas/issues/72) 

# Comments:
`all_contrasts` and `case_vs_group_contrasts` require high memory so it makes difficult CI tests. Removed from CI tests until the problem is addressed specifically. Card opened: https://github.com/lifebit-ai/gel-gwas/issues/71

# Test in CloudOS
https://cloudos.lifebit.ai/app/jobs/5f9ace9a61aba101120efbe3

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,33 +3,143 @@ name: tests for lifebit-ai/gel-gwas
...
-          nextflow run ${GITHUB_WORKSPACE} -profile test_binary_min_none_case_vs_control_contrast_genetic_correlation_h2
-  test_gwas_catalogue:
+          nextflow run ${GITHUB_WORKSPACE} --design_mode ${{ matrix.design_mode }} --continuous_var_transformation ${{ matrix.continuous_var_transformation }}  --continuous_var_aggregation ${{ matrix.continuous_var_aggregation }} -profile test_binary
+  test_binary_h2:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         nxf_ver: ['20.01.0', '']
+        continuous_var_aggregation: [median]
+        continuous_var_transformation: [None]
```
Comment (cgpu) [inline .github/workflows/ci.yml:30]: We can omit some extra profiles if we add all possible values in an array:

```Suggestion
        continuous_var_aggregation: [median, mean, min, max]
        continuous_var_transformation: [ None, log, log10, log2, zscores]
```

File: main.nf [hunk 2]
```diff
@@ -181,7 +179,7 @@ if (params.trait_type == 'binary'){
 
     output:
     set val(name), val(chr), file("${name}.filtered_final.vcf.gz"), file("${name}.filtered_final.vcf.gz.csi") into filteredVcfsCh
-    file("${name}_filtered.{bed,bim,fam}") into plinkTestCh
+    // file("${name}_filtered.{bed,bim,fam}") into plinkTestCh
```
Comment (cgpu) [inline main.nf:182]: Should we delete or add back in the script?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -3,33 +3,143 @@ name: tests for lifebit-ai/gel-gwas
...
+        design_mode: ['case_vs_control_contrast']
     steps:
       - uses: actions/checkout@v1
```
Comment (cgpu) [inline .github/workflows/ci.yml:33]:
```Suggestion
      - uses: actions/checkout@v2
```
Updating actions version, latest is now v2.

File: main.nf (hunk 2 above)
Comment (mcamarad) [reply main.nf:182]: I think deleting is the best option, committing a deletion of this

**Comment by mcamarad** [comment]
Thanks @cgpu !! Merging now and putting this in quarantine until tests and future fixes are done ✅

**Pull request merged**